aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -706,6 +706,19 @@ Tensor linspace(
   return at::linspace_out(result, start, end, steps);
 }
 
+Tensor linspace(
+    const Tensor& start,
+    const Tensor& end,
+    int64_t steps,
+    c10::optional<ScalarType> dtype,
+    c10::optional<Layout> layout,
+    c10::optional<Device> device,
+    c10::optional<bool> pin_memory) {
+  TORCH_CHECK(start.dim() == 0 && end.dim() == 0, "linspace only supports 0-dimensional start and end tensors, "
+    "but got start with ", start.dim(), " dimension(s) and end with ", end.dim()," dimension(s).");
+  return at::linspace(start.item(), end.item(), steps, dtype, layout, device, pin_memory);
```

Comment:
Can you add a quick test somewhere on what happens if `start` or `end` has requires_grad=True? The output of linspace should not be a part of the computation graph and should not require grad.

Reply:
They are end up converted to scalars before passed into the kernel so whether they require grad doesn't matter I think.

Or, do you mean adding an assertion like `TORCH_CHECK(!start.requires_grad() && !end.requires_grad(), "linspace(): The start and end tensors should not require grad.");`?

Reply:
Oh no, what I meant was a test to check if the inputs of linspace have requires_grad=True, then the output of linspace does not have requires_grad=True

Reply:
Added test in `test/test_torch.py`